src/e2e/java/teammates/e2e/cases/e2e/InstructorCoursesPageE2ETest.java

```diff
@@ -0,0 +1,192 @@
+package teammates.e2e.cases.e2e;
+
+import java.time.Instant;
+import java.time.ZoneId;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.CourseAttributes;
+import teammates.common.util.AppUrl;
+import teammates.common.util.Const;
+import teammates.common.util.ThreadHelper;
+import teammates.e2e.pageobjects.AppPage;
+import teammates.e2e.pageobjects.InstructorCoursesPage;
+import teammates.e2e.pageobjects.InstructorHomePage;
+import teammates.e2e.util.BackDoor;
+
+/**
+ * SUT: {@link Const.WebPageURIs#INSTRUCTOR_COURSES_PAGE}.
+ */
+public class InstructorCoursesPageE2ETest extends BaseE2ETestCase {
+    CourseAttributes[] courses = new CourseAttributes[3];
+    CourseAttributes newCourse;
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/InstructorCoursesPageE2ETest.json");
+        removeAndRestoreDataBundle(testData);
+
+        courses[0] = testData.courses.get("CS1101");
+        courses[1] = testData.courses.get("CS2104");
+        courses[2] = testData.courses.get("CS2105");
+
+        newCourse = CourseAttributes.builder("ICAddE2ETest.CS4100")
+                .withName("New Course")
+                .withTimezone(ZoneId.of("Asia/Singapore"))
+                .build();
+    }
+
+    @BeforeClass
+    public void classSetup() {
+        BackDoor.deleteCourse(newCourse.getId());
+    }
+
+    @Test
+    public void testAll() {
+        String instructorId = testData.accounts.get("instructor").getGoogleId();
+        AppUrl url = createUrl(Const.WebPageURIs.INSTRUCTOR_COURSES_PAGE)
+                .withUserId(instructorId);
+        loginAdminToPage(url, InstructorHomePage.class);
+        InstructorCoursesPage coursesPage = AppPage.getNewPageInstance(browser, url, InstructorCoursesPage.class);
+        coursesPage.waitForPageToLoad();
+
+        ______TS("verify loaded data");
+        CourseAttributes[] activeCourses = { courses[0] };
+        CourseAttributes[] archivedCourses = { courses[1] };
+        CourseAttributes[] deletedCourses = { courses[2] };
+
+        coursesPage.verifyActiveCoursesDetails(activeCourses);
+        coursesPage.verifyArchivedCoursesDetails(archivedCourses);
+        coursesPage.verifyDeletedCoursesDetails(deletedCourses);
+
+        ______TS("verify statistics");
+        coursesPage.verifyActiveCourseStatistics(courses[0], "1", "3", "3", "1");
```

Comment:
This is a minor nit - when you hardcode in "1" / "3" / "3" / "1", you may run into problems when the course details are modified (you have to change two places instead of one). Perhaps getting it programmatically might be better?